kyotocabinet: New package

EbiArnie opened the pull request:
This adds Kyoto Cabinet, an in-process database manager

EbiArnie committed:
kyotocabinet: New package

EbiArnie committed:
Moved to correct dir

alalazo commented on var/spack/repos/builtin/packages/kyotocabinet/package.py:
```diff
@@ -0,0 +1,20 @@
+
+    version("1.2.80", sha256="4c85d736668d82920bfdbdb92ac3d66b7db1108f09581a769dd9160a02def349")
+
+    depends_on("zlib@1.2.3:", type=("build", "link"))
```
I think this ought to be:
```suggestion
    depends_on("zlib-api@1.2.3:", type=("build", "link"))
```
for better composability with other packages.

EbiArnie replied:
OK

alalazo commented on var/spack/repos/builtin/packages/perl-class-tiny/package.py:
```diff
@@ -0,0 +1,20 @@
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack.package import *
```
This seems to be unrelated to this PR, and a duplicate of #41513

EbiArnie replied:
Indeed, sorry about that.

EbiArnie committed:
Depend on zlib-api, remove unrelated package

For kyotocabinet, depend on zlib-api instead of zlib.
Remove perl-class-tiny, this was added in error
